### ISSUE TITLE
Consume GitHub releases

### ISSUE THREAD

**Issue by migmartri** [chainloop-dev/chainloop#785]
We want to be able to attest a gitHub release in an easy way. 

Ideally, we could have a reusable workflow/action that reacts to a `release` event and performs an attestation of all the published artifacts.

As explained here https://github.com/chainloop-dev/chainloop/pull/772#discussion_r1599939348 we can download all the artifacts using the GH CLI, and then we could push them to Chainloop by using a `CHAINLOOP_TOKEN`

We could even automatically create a workflow if needed using also our new reusable workflow for that purpose cc/ @jiparis 

```[tasklist]
### Tasks
- [ ] https://github.com/chainloop-dev/chainloop/issues/786
- [ ] https://github.com/chainloop-dev/chainloop/issues/816
- [ ] https://github.com/chainloop-dev/chainloop/issues/788
- [ ] https://github.com/chainloop-dev/chainloop/issues/787
- [ ] https://github.com/chainloop-dev/chainloop/issues/844
- [x] add guard to check it's a release
- [ ] https://github.com/chainloop-dev/chainloop/issues/849
- [ ] https://github.com/chainloop-dev/chainloop/issues/852
- [ ] Be able to configure the output of contract-less materials
- [x] Modify existing release contracts?
- [ ] https://github.com/chainloop-dev/chainloop/issues/866
- [ ] https://github.com/chainloop-dev/chainloop/issues/867
- [ ] https://github.com/chainloop-dev/chainloop/issues/868
```

**Comment by javirln** [comment]
One of the issues found when deploying the reusable workflow into our pipelines is the fact that a workflow cannot be triggered by other workflow if the token used is the default GitHub token. That provoked that the `build and package` workflow didn't the `release` workflow since, what triggers a release within `build and package` is `goreleaser` using a GitHub Token.

More information regarding the issue can be found here: https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow

This does not mean the workflow does not work, it only means that in order to be triggered by another workflow it needs to use a personal access token, manual releases of course trigger the workflow.

**Comment by migmartri** [comment]
Should we support workflow dispatch? That one seems to be supported.

**Comment by javirln** [comment]
Additionally to the `release` trigger? 
How do you think it would be the flow for our use case? Once `build and package` is finished, us manually going to the workflow and trigger it? In that case we need to check that the github.ref_name is still the tag

**Comment by migmartri** [comment]
basically, the reusable workflow uses a tag as input. Then this workflow can be used from different parent jobs, either by being triggered explicitly or in a workflow that reacts to the `release` event.

Either way, the reusable workflow is the same but gets triggered/loaded in two different contexts, does it make sense?

**Comment by javirln** [comment]
Closing this task in favor of its follow up one: https://github.com/chainloop-dev/chainloop/issues/869

Additionally the following issues should be considered once tackling that task since **they were not part of this one**:
- https://github.com/chainloop-dev/chainloop/issues/866
- https://github.com/chainloop-dev/chainloop/issues/867
- https://github.com/chainloop-dev/chainloop/issues/868
